drivers/ng_at86rf2xx/ng_at86rf2xx_getset.c

```diff
@@ -334,43 +334,68 @@ void ng_at86rf2xx_set_option(ng_at86rf2xx_t *dev, uint16_t option, bool state)
     }
 }
 
-uint8_t ng_at86rf2xx_get_state(ng_at86rf2xx_t *dev)
+static inline void _set_state(ng_at86rf2xx_t *dev, uint8_t state)
 {
-    uint8_t status = ng_at86rf2xx_get_status(dev);
-    return (status & 0x1f);
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__TRX_STATE, state);
+    while (ng_at86rf2xx_get_status(dev) != state);
 }
 
-static inline void _set_state(ng_at86rf2xx_t *dev, uint8_t state)
+static inline void _force_trx_off(ng_at86rf2xx_t *dev)
 {
-    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__TRX_STATE, state);
-    while (ng_at86rf2xx_get_state(dev) != state);
+    ng_at86rf2xx_reg_write(dev, NG_AT86RF2XX_REG__TRX_STATE, NG_AT86RF2XX_TRX_STATE__FORCE_TRX_OFF);
+    while (ng_at86rf2xx_get_status(dev) != NG_AT86RF2XX_STATE_TRX_OFF);
 }
 
 void ng_at86rf2xx_set_state(ng_at86rf2xx_t *dev, uint8_t state)
 {
-    uint8_t old_state = ng_at86rf2xx_get_state(dev);
+    uint8_t old_state = ng_at86rf2xx_get_status(dev);
```

Comment:
Originally it was my intention to use `ng_at86rf2xx_t.state` to keep track to minimize SPI transactions and map to changeable states. We have to keep in mind `ng_at86rf2xx_get_status` can also return `BUSY_RX_AACK` or similar.
This can be a problem in several places in this driver and we have to find a way to deal with it.